src/main/java/core/basesyntax/Injector.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoSuchImplementationException {
+        Constructor declaredConstructor = clazz.getDeclaredConstructor();
+        Object newInstance = declaredConstructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        Class<BetDaoImpl> betDaoClass = BetDaoImpl.class;
+        Class<UserDaoImpl> userDaoClass = UserDaoImpl.class;
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(BetDao.class)) {
+                field.setAccessible(true);
+                if (betDaoClass.isAnnotationPresent(Dao.class)) {
+                    field.set(newInstance, Factory.getBetDaoFactory());
+                } else {
+                    throw new NoSuchImplementationException("We don't have"
+                            + " implementation of BetDao");
+                }
+            }
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(UserDao.class)) {
+                field.setAccessible(true);
+                if (userDaoClass.isAnnotationPresent(Dao.class)) {
+                    field.set(newInstance, Factory.getUserDaoFactory());
+                } else {
+                    throw new NoSuchImplementationException("We don't have"
+                            + " implementation of UserDao");
+                }
+            }
```

Comment:
let's try to simplify this part

let's have first `if` checking if field has `Inject` annotation

then inside that `if`:
1) check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2) else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3) else - throw Exception

Reply:
done